plugin/formatting.py

```diff
@@ -37,20 +37,22 @@ def is_applicable(cls, view_settings: dict) -> bool:
         return False
 
     def on_pre_save(self) -> None:
-        if self.view.settings().get("lsp_format_on_save"):
-            client = client_from_session(session_for_view(self.view, 'documentFormattingProvider'))
-            if client:
-                params = {
-                    "textDocument": {
-                        "uri": filename_to_uri(self.view.file_name())
-                    },
-                    "options": options_for_view(self.view)
-                }
-                request = Request.formatting(params)
-                response = client.execute_request(request)
-                if response:
-                    apply_response_to_view(response, self.view)
-                    global_events.publish("view.on_purge_changes", self.view)
+        file_path = self.view.file_name()
+        if file_path:
+            if self.view.settings().get("lsp_format_on_save"):
```

Comment:
I don't know what's the style of this package but my preference would be to create as little nesting as possible. So either merge those two `if`'s or even return early instead of nesting.

Feel free to totally ignore me. :)

Reply:
Good point, changed it!